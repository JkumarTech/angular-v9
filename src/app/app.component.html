<mat-sidenav-container class="sidenav-container">
  <mat-sidenav #drawer class="sidenav" [fixedInViewport]="true">
    <mat-toolbar>Menu</mat-toolbar>
    <mat-nav-list>
      <a
        mat-list-item
        i18n="route: people|Link to people route"
        routerLink="/people"
        >People</a
      >
      <a
        mat-list-item
        i18n="route: cantina|Link the Mos Eisley Cantina route"
        routerLink="/mos-eisley/cantina"
        >Mos Eisley Cantina</a
      >
    </mat-nav-list>
  </mat-sidenav>
  <mat-sidenav-content class="sidenav-content">
    <mat-toolbar color="primary">
      <button
        type="button"
        aria-label="Toggle sidenav"
        mat-icon-button
        (click)="drawer.toggle()"
      >
        <mat-icon aria-label="Side nav toggle icon">menu</mat-icon>
      </button>
      <span>StrWrs</span>
    </mat-toolbar>
    <div class="content">
      <router-outlet></router-outlet>
    </div>
  </mat-sidenav-content>
</mat-sidenav-container>
