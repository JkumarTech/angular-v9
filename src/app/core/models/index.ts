export * from './film.interface';
export * from './person.interface';
export * from './planet.interface';
