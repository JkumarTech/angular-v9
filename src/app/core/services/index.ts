export * from './film.service';
export * from './person.service';
export * from './planet.service';
export * from './window.service';
