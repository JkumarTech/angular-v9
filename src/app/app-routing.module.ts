import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { HomeComponent } from './home';

const routes: Routes = [
  {
    path: '',
    component: HomeComponent
  },
  {
    path: 'mos-eisley',
    loadChildren: () =>
      import('./features/mos-eisley/mos-eisley.module').then(
        ({ MosEisleyModule }) => MosEisleyModule
      )
  },
  {
    path: 'people',
    loadChildren: () =>
      import('./features/people/people.module').then(
        ({ PeopleModule }) => PeopleModule
      )
  },
  {
    path: 'recon',
    loadChildren: () =>
      import('./features/recon/recon.module').then(
        ({ ReconModule }) => ReconModule
      )
  },
  {
    path: '**',
    redirectTo: ''
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
