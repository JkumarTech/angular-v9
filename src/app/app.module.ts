import { CommonModule } from '@angular/common';
import { registerLocaleData } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import localeDe from '@angular/common/locales/de';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { WINDOW_PROVIDERS } from './core';
import { HomeComponent } from './home';
import { MaterialModule } from './material';
import { SharedModule } from './shared';

registerLocaleData(localeDe);

@NgModule({
  declarations: [AppComponent, HomeComponent],
  imports: [
    AppRoutingModule,
    BrowserAnimationsModule,
    BrowserModule,
    CommonModule,
    HttpClientModule,
    MaterialModule,
    SharedModule
  ],
  providers: [WINDOW_PROVIDERS],
  bootstrap: [AppComponent]
})
export class AppModule {}
