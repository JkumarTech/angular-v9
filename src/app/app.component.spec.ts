import { CommonModule } from '@angular/common';
import { ComponentFixture, async } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';

import { ConfigureFn, configureTests } from '../test-config.helper';
import { AppComponent } from './app.component';
import { MaterialModule } from './material';
import { SharedModule } from './shared';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let app: AppComponent;

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [AppComponent],
        imports: [
          CommonModule,
          NoopAnimationsModule,
          RouterTestingModule,
          MaterialModule,
          SharedModule
        ]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(AppComponent);
      app = fixture.componentInstance;
      fixture.detectChanges();
    });
  }));

  it('should create the app', async(() => {
    expect(app).toBeTruthy();
  }));

  it('should render html', async(() => {
    const compiled = fixture.debugElement.nativeElement;
    // @ts-ignore
    expect(compiled).toMatchSnapshot();
  }));
});
