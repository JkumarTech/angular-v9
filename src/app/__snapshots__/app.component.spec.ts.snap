// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render html 1`] = `
<div
  id="root1"
  ng-version="9.0.0-rc.6"
>
  <mat-sidenav-container
    class="sidenav-container mat-drawer-container mat-sidenav-container"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="sidenav mat-drawer mat-sidenav ng-tns-c7-0 ng-trigger ng-trigger-transform mat-drawer-over mat-sidenav-fixed ng-star-inserted"
      fixedinviewport="true"
      ng-reflect-fixed-in-viewport="true"
      style="top:0px;bottom:0px;box-shadow:none;visibility:hidden;0:top;1:bottom;2:box-shadow;3:visibility;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      >
        <mat-toolbar
          class="mat-toolbar mat-toolbar-single-row"
        >
          Menu
        </mat-toolbar>
        <mat-nav-list
          class="mat-nav-list mat-list-base"
          role="navigation"
        >
          <a
            class="mat-list-item"
            href="/people"
            mat-list-item=""
            ng-reflect-router-link="/people"
            routerlink="/people"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="https://github.com/people"
              />
              <div
                class="mat-list-text"
              />
              People
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/mos-eisley/cantina"
            mat-list-item=""
            ng-reflect-router-link="/mos-eisley/cantina"
            routerlink="/mos-eisley/cantina"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="https://github.com/mos-eisley/"
              />
              <div
                class="mat-list-text"
              />
              Mos Eisley Cantina
            </div>
          </a>
        </mat-nav-list>
      </div>
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="sidenav-content mat-drawer-content mat-sidenav-content"
    >
      <mat-toolbar
        class="mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
        ng-reflect-color="primary"
      >
        <button
          aria-label="Toggle sidenav"
          class="mat-icon-button mat-button-base _mat-animation-noopable"
          mat-icon-button=""
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            <mat-icon
              aria-hidden="true"
              aria-label="Side nav toggle icon"
              class="mat-icon notranslate material-icons mat-icon-no-color"
              role="img"
            >
              menu
            </mat-icon>
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <span>
          StrWrs
        </span>
      </mat-toolbar>
      <div
        class="content"
      >
        <router-outlet />
      </div>
    </mat-sidenav-content>
    
  </mat-sidenav-container>
</div>
`;
